rules/select_file.bzl

```diff
@@ -0,0 +1,33 @@
+"""
+select_file() build rule implementation.
+
+Selects a single file from the outputs of some target by given relative path.
+"""
+
+def _impl(ctx):
+    if ctx.attr.subpath and len(ctx.attr.subpath) == 0:
+        fail("Subpath can not be empty.")
+
+    out = None
+    canonical = ctx.attr.subpath.replace("\\", "/")
```

Comment:
do you know if we have common functionalities to deal with OS-specific path issues like this?

Reply:
Not that I know of.